apps/ecommerce/frontend/src/components/ResourceCardMenu.tsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from 'react';
+import { IconButton, Menu } from '@contentful/f36-components';
+import { MoreHorizontalIcon } from '@contentful/f36-icons';
+
+interface Props {
+  onRemove: Function;
+  isDataVisible: boolean;
+  onToggleDataVisible: Function;
+  index?: number;
+  total?: number;
+  onMoveToTop?: Function;
+  onMoveToBottom?: Function;
+}
+
+const ResourceCardMenu = (props: Props) => {
+  const {
+    onRemove,
+    isDataVisible,
+    onToggleDataVisible,
+    index,
+    total,
+    onMoveToTop,
+    onMoveToBottom,
+  } = props;
+
+  const [menuItems, setMenuItems] = useState([
+    <Menu.Item key="copy" onClick={() => onRemove()}>
+      Remove
+    </Menu.Item>,
+  ]);
+
+  useEffect(() => {
+    if (typeof index !== 'undefined' && total && total > 1) {
+      if (index > 0 && onMoveToTop) {
+        menuItems.push(
+          <Menu.Item key="moveToTop" onClick={() => onMoveToTop}>
+            Move to top
+          </Menu.Item>
+        );
+      }
+
+      if (total !== index + 1 && onMoveToBottom) {
+        menuItems.push(
+          <Menu.Item key="moveToBottom" onClick={() => onMoveToBottom}>
+            Move to bottom
+          </Menu.Item>
+        );
+      }
+    }
+
+    menuItems.push(<Menu.Divider key="divider" />);
+    menuItems.push(
+      <Menu.Item key="toggleData" onClick={() => onToggleDataVisible()}>
+        {isDataVisible ? 'Hide' : 'Show'} Data
+      </Menu.Item>
+    );
+
+    setMenuItems([...menuItems]);
+  }, []);
```

Comment:
Feel free to change the array here for states to listen to for re-rendering. Or nothing at all if it just needs to be rendered once on load